shared/src/current_block.rs

```diff
@@ -156,6 +159,18 @@ where
             .context("failed to get current block")?
             .as_u64())
     }
+
+    async fn block_number_from_tx_hash(&self, hash: H256) -> Result<u64> {
+        let potential_receipt = self.eth().transaction_receipt(hash).await?;
+        if let Some(receipt) = potential_receipt {
+            // Need to unwrap Option<U64> to convert U64 -> u64
+            if let Some(block_number) = receipt.block_number {
+                return Ok(block_number.as_u64());
+            }
+            return Err(anyhow!("no block number with transaction receipt"));
+        }
+        return Err(anyhow!("no transaction receipt found"));
+    }
```

Comment:
This can be simplified using `Option::ok_or_else` which promotes the `Option` to a `Result` by specifying an error for `None`:

```suggestion
        Ok(self.eth()
            .transaction_receipt(hash)
            .await?
            .ok_or_else(|| anyhow!("no transaction receipt found"))?
            .block_number
            .ok_or_else(|| anyhow("no block number with transaction receipt"))?
            .as_u64())
```

Reply:
I knew there was something nicer that could be done!